packages/viz/src/lib/Layer.ts

```diff
@@ -146,6 +150,110 @@ export class Layer {
   public get source() {
     return this._source;
   }
+
+  /**
+   * @public
+   * This method creates popups every time the
+   * user clicks on one or more features of the layer.
+   */
+  public async setPopupClick(elements: PopupElement[] | string[] | null = []) {
+    if (elements && elements.length > 0) {
+      let popup: Popup | null = null;
+      // eslint-disable-next-line padding-line-between-statements, @typescript-eslint/no-unused-vars
+      this._options.onClick = info => {
+        if (this._deckInstance) {
+          const {
+            lngLat,
+            object: { properties }
+          } = info;
+
+          const popupContent: string = generatePopupContent(
+            elements,
+            properties
+          );
+
+          if (popup === null) {
+            popup = new Popup();
+          }
+
+          popup.setContent(popupContent);
+          popup.setCoordinate(lngLat);
+          popup.addTo(this._deckInstance);
+        }
+      };
+
+      this._options.pickable = true;
+    } else {
+      this._options.onClick = undefined;
+      this._options.pickable = !!this._options.onHover;
```

Comment:
Be careful with this. If we remove pickability from a layer, highlighting is not going to work if enabled. So we shouldn't change this, and leave it always to the default or the value that the user defined it to.